ports/stm/supervisor/port.c

```diff
@@ -106,3 +140,109 @@ void HardFault_Handler(void) {
         asm("nop;");
     }
 }
+
+// This function is called often for timing so we cache the seconds elapsed computation based on the
+// register value. The STM HAL always does shifts and conversion if we use it directly.
+volatile uint32_t seconds_to_date = 0;
+volatile uint32_t cached_date = 0;
+volatile uint32_t seconds_to_minute = 0;
+volatile uint32_t cached_hours_minutes = 0;
+uint64_t port_get_raw_ticks(uint8_t* subticks) {
```

Comment:
frankly this is horrifying.  but necessary.  This "RTC" will never actually store real time, but will just count from some power-on value up forever, and be corrected to real time in the shared code.

Reply:
I wouldn't say never. I can see a world where we actually suspend everything but the RTC. In that case we'll want to save the offset through the RTC and up date it to correct time.

I don't ever plan to interface it through a date API though. It doesn't make sense given most RTCs are counters and not clocks.